### PR TITLE
Upgrade to reqwest 0.9

### LINKED ISSUE #453: Recipe: Creating Database in File System
|  |  |
| ------------- | ------------- |
| Crates  | rusqlite|
| Section  | 7.1 SQLite  |

This recipe should demonstrate using rusqlite by saving the database to the file system.

### LINKED ISSUE #459: Bump reqwest to 0.9
`reqwest 0.9` was recently released, using `hyper 0.12`. A couple of examples need to be updated because of the header changes. It should also help with `openssl` compatibility -- I can't build the examples because of a version mismatch.

I'll try to work on this.

### PR DESCRIPTION
**Pull request by lnicola**
fixes #453, #454

Based on #460. I'll rebase afterwards, please ignore the first two commits.

CC @robertDurst as per #455.

--------------------------
### Things to check before submitting a PR

- [x] the tests are passing locally with `cargo test`
- [x] cookbook renders correctly in `mdbook serve -o`
- [x] commits are squashed into one and rebased to latest master
- [x] PR contains correct "fixes #ISSUE_ID" clause to autoclose the issue on PR merge
- [x] spell check runs without errors `./ci/spellchecker.sh`
- [x] link check runs without errors `link-checker ./book`
- [x] non rendered items are in sorted order (links, reference, identifiers, Cargo.toml)
- [x] links to docs.rs have wildcard version `https://docs.rs/tar/*/tar/struct.Entry.html`
- [x] example has standard [error handling](https://rust-lang-nursery.github.io/rust-cookbook/about.html#a-note-about-error-handling)
- [x] code identifiers in description are in hyperlinked backticks

### Things to do after submitting PR
- [ ] check if CI is happy with your PR

### CODE REVIEW

File: src/web/clients/download/partial.md [hunk 1]
```diff
@@ -34,32 +36,29 @@ use reqwest::StatusCode;
...
+#     type Item = HeaderValue;
 #
 #     fn next(&mut self) -> Option<Self::Item> {
 #         if self.start > self.end {
 #             None
 #         } else {
 #             let prev_start = self.start;
 #             self.start += std::cmp::min(self.buffer_size as u64, self.end - self.start + 1);
-#             Some(Range::bytes(prev_start, self.start - 1))
+#             Some(HeaderValue::from_str(&format!("bytes={}-{}", prev_start, self.start - 1)).unwrap())
```
Comment (AndyGauge) [inline src/web/clients/download/partial.md]: Can we avoid `unwrap()`?  We are really trying to teach how to implement these functions in production code, to handle runtime errors gracefully.

File: src/web/clients/download/partial.md [hunk 2]
```diff
@@ -2,28 +2,30 @@
...
 [![reqwest-badge]][reqwest] [![cat-net-badge]][cat-net]
 
-Uses [`reqwest::Client::head`] to get the content-length and validates the
-server setting the header [`reqwest::header::ContentRange`].
+Uses [`reqwest::Client::head`] to get the [Content-Length] of the response.
 
-If supported, downloads the content using [`reqwest::get`] by setting the
-[`reqwest::header::Range`] to do partial downloads printing basic progress
-messages in chunks of 10240 bytes.
+If supported, downloads the content using [`reqwest::Client::get`] and
```
Comment (AndyGauge) [inline src/web/clients/download/partial.md]: It took a few re-reads for me to get what was 'setting the Range header'.  Can we go with: 
If supported, downloads the content using reqwest::Client::get with the Range header set to do partial ...

File: src/web/clients/download/partial.md (hunk 1 above)
Comment (lnicola) [reply src/web/clients/download/partial.md]: I don't think so. The error case is for things like non-ASCII characters, I suppose. That can be proven statically (`prev_start` and `self.start - 1` are integers, and the format string is fine), but we can't avoid the `unwrap()` until it gets hidden away in the `headers` crate.

File: src/web/mime/request.md [hunk 3]
```diff
@@ -3,38 +3,44 @@
 [![reqwest-badge]][reqwest] [![mime-badge]][mime] [![cat-net-badge]][cat-net] [![cat-encoding-badge]][cat-encoding]
 
 When receiving a HTTP reponse from *reqwest* the [MIME type] or media type may be
```
Comment (AndyGauge) [inline src/web/mime/request.md]: I made a mistake when I merged the crate names Italicized.  Can you replace *reqwest* with `reqwest` and *mime* with `mime` in this PR?

File: src/web/clients/download/partial.md (hunk 2 above)
Comment (lnicola) [reply src/web/clients/download/partial.md]: To be fair, the old phrasing wasn't much better :-).

Does this sound all right?

> The code then uses [`reqwest::Client::get`] to download the content in
chunks of 10240 bytes, while printing progress messages. The requested
ranges are specified using with the [Range] header.

**Comment by lnicola** [comment]
Added a new commit with some phrasing changes. I'll squash after you review it.

File: src/web/clients/download/partial.md (hunk 1 above)
Comment (lnicola) [reply src/web/clients/download/partial.md]: Added a comment to explain why the `unwrap` never fails.

**Comment by lnicola** [comment]
Rebased and squashed.

**Pull request merged**